src/compas_view2/objects/object.py

```diff
@@ -34,3 +64,50 @@ def create(self):
 
     def edit(self):
         pass
+
+    @property
+    def translation(self):
+        return self._translation
+
+    @translation.setter
+    def translation(self, data):
+        self._translation = data
+
+    @property
+    def rotation(self):
+        return self._rotation
+
+    @rotation.setter
+    def rotation(self, data):
+        self._rotation = data
+
+    @property
+    def scale(self):
+        return self._scale
+
+    @scale.setter
+    def scale(self, data):
+        self._scale = data
+
+    def _update_matrix(self):
+        """Update the matrix from object's translation, rotation and scale"""
+        T1 = Translation.from_vector(self.translation)
+        R1 = Rotation.from_euler_angles(self.rotation)
+        S1 = Scale.from_factors(self.scale)
+        M = T1 * R1 * S1
+        self._transformation.data = M.data
```

Comment:
not sure if this is the most efficient way to go about this

Reply:
You mean the method to compose the matrix or the general behavior of keep updating the matrix?

Reply:
both :) but not important now...

the data thing seems wrong though. why not just do `self._transformation = M` or `self._transformation.matrix = M.matrix`?

although also the latter seems redundant...

Reply:
yep was looking at the code for matrix...... I wanted to make sure it is only data copied without replacing the entire object. And realised the data setter does that anyway....